File: app/src/main/java/com/aib/net/Resource.java
package com.aib.net;

public class Resource<T> {
    public String msg;
    public Status status;
    public T data;

    public Resource(String msg, Status status, T data) {
        this.msg = msg;
        this.status = status;
        this.data = data;
    }

    public static <T> Resource loading(String msg, T data) {
        return new Resource<T>(msg, Status.LOADING, data);
    }

    public static <T> Resource success(String msg, T data) {
        return new Resource<T>(msg, Status.SUCCESS, data);
    }

    public static <T> Resource error(String msg, T data) {
        return new Resource<T>(msg, Status.ERROR, data);
    }
}

File: app/src/main/java/com/aib/net/Resource.kt
package com.aib.net

class Resource<T>(var msg: String?, var status: Status, var data: T?) {
    companion object {

        /**
         * 请求前
         */
        fun <T> loading(): Resource<T> {
            return Resource("数据加载中...", Status.LOADING, null)
        }

        /**
         * 请求成功
         */
        fun <T> success(data: T): Resource<T> {
            return Resource(null, Status.SUCCESS, data)
        }

        /**
         * 请求失败
         */
        fun <T> error(msg: String): Resource<T> {
            return Resource(msg, Status.ERROR, null)
        }


    }
}
